src/Config/TokenConfig.ts

```diff
@@ -0,0 +1,131 @@
+import {Validator} from "jsonschema";
+const schema = require('./TokenConfig.schema.json');
+const path = require('path');
+
+/**
+ * Contract addresses of the origin chain specific to a token.
+ */
+export class OriginContracts {
+  public baseTokenAddress: Address;
+
+  public valueTokenAddress: Address;
+
+  public gatewayOrganizationAddress: Address;
+
+  public eip20GatewayAddress: Address;
+
+  public gatewayLibAddress?: Address;
+
+  public messageBusAddress?: Address;
+
+  public merklePatriciaLibAddress?: Address;
+
+  public stakePoolAddress?: Address;
+}
+
+/**
+ * Contract addresses deployed on the auxiliary chain specific to a token.
+ */
+export class AuxiliaryContracts {
+  public coGatewayOrganizationAddress: Address;
+
+  public utilityTokenAddress: Address;
+
+  public eip20CoGatewayAddress: Address;
+
+  public gatewayLibAddress?: Address;
+
+  public messageBusAddress?: Address;
+
+  public merklePatriciaLibAddress?: Address;
+
+  public redeemPoolAddress?: Address;
+}
+
+/**
+ * Holds the config of a deployed token.
+ */
+export default class TokenConfig {
+  public mosaicConfig: MosaicConfig;
+
+  public auxChainId: number;
+
+  public originContracts: OriginContracts;
+
+  public auxiliaryContracts: AuxiliaryContracts;
+
+  private constructor(config: any) {
+
+    this.mosaicConfig = MosaicConfig.fromFile(config.mosaicConfigFilePath);
+    this.auxChainId = config.auxChainId;
+    this.originContracts = config.originContracts;
+    this.auxiliaryContracts = config.auxiliaryContracts;
+  }
+
+  /**
+   * @param {string} filePath tokenConfig absolute path.
+   *
+   * @return {TokenConfig} TokenConfig object.
+   */
+  public static fromFile(filePath: string): TokenConfig {
+    if (fs.existsSync(filePath)) {
+      const configObject = TokenConfig.readConfigFromFile(filePath);
+      return new TokenConfig(configObject);
+    }
+    throw new TokenConfigNotFoundException(`Missing config file at path: ${filePath}`);
+  }
+
+  /**
+   * Read config from file, validate it and return as JSON object.
+   *
+   * @param {string} filePath tokenConfig absolute path.
+   *
+   * @return {object} Json parsed object.
+   */
+  private static readConfigFromFile(filePath: string): object {
+    const configString = fs.readFileSync(filePath).toString();
+    if (configString && configString.length > 0) {
+      const configObject = JSON.parse(configString);
+      TokenConfig.validateSchema(configObject);
+      configObject.mosaicConfigFilePath = TokenConfig.resolveHomePath(
+        configObject.mosaicConfigFilePath
+      );
+      return configObject;
+    }
+    throw new InvalidTokenConfigException(`blank config file found at: ${filePath}`);
+  }
+
+  /**
+   * It resolves token config file path to absolute path after resolving tilde.
+   *
+   * @param filePath Token config file path. e.g. ~/.mosaic/goerli/WETH.json
+   * @return {string} Absolute file path.
+   */
+  private static resolveHomePath(filePath: string) {
```

Comment:
This logic should be moved to other class may be `FileSystem` or `Directory`

Reply:
Done!